src/components/NewCommentForm.tsx

```diff
@@ -1,9 +1,139 @@
-import React from 'react';
+import React, { useEffect, useState } from 'react';
+import { newComment } from '../api/newComment';
+import { Comment } from '../types/Comment';
+import classNames from 'classnames';
+import { Errors } from '../types/Errors';
+
+type Props = {
+  postId: number;
+  setComments: React.Dispatch<React.SetStateAction<Comment[]>>;
+};
+
+export const NewCommentForm: React.FC<Props> = ({ postId, setComments }) => {
+  const [currentName, setCurrentName] = useState('');
+  const [currentEmail, setCurrentEmail] = useState('');
+  const [currentBody, setCurrentBody] = useState('');
+  const [isAddingComment, setIsAddingComment] = useState(false);
+  const [isNameError, setIsNameError] = useState(false);
+  const [isEmailError, setIsEmailError] = useState(false);
+  const [isBodyError, setIsBodyError] = useState(false);
+  const [currentError, setCurrentError] = useState<Errors | null>(null);
+
+  const resetForm = (reason: 'submit' | 'clear') => {
+    if (reason === 'clear') {
+      setCurrentName('');
+      setCurrentEmail('');
+    }
+
+    setCurrentBody('');
+    setIsNameError(false);
+    setIsEmailError(false);
+    setIsBodyError(false);
+  };
+
+  const handleNameChange = (event: React.ChangeEvent<HTMLInputElement>) => {
+    setCurrentName(event.target.value);
+    if (isNameError) {
+      setIsNameError(false);
+    }
+  };
+
+  const handleEmailChange = (event: React.ChangeEvent<HTMLInputElement>) => {
+    setCurrentEmail(event.target.value);
+    if (isEmailError) {
+      setIsEmailError(false);
+    }
+  };
+
+  const handleBodyChange = (event: React.ChangeEvent<HTMLTextAreaElement>) => {
+    setCurrentBody(event.target.value);
+    if (isBodyError) {
+      setIsBodyError(false);
+    }
+  };
+
+  const validateName = (testedName: string) => {
+    if (testedName.trim() === '') {
+      setIsNameError(true);
+    }
+
+    return testedName.trim().length > 0;
+  };
+
+  const validateEmail = (testedEmail: string) => {
+    if (!/\S+@\S+\.\S+/.test(testedEmail)) {
+      setIsEmailError(true);
+    }
+
+    return /\S+@\S+\.\S+/.test(testedEmail);
+  };
+
+  const validateCommentBody = (testedBody: string) => {
+    if (testedBody.trim() === '') {
+      setIsBodyError(true);
+    }
+
+    return testedBody.trim().length > 0;
+  };
+
+  const validateForm = () => {
+    const nameValid = validateName(currentName);
+    const emailValid = validateEmail(currentEmail);
+    const bodyValid = validateCommentBody(currentBody);
+
+    return nameValid && emailValid && bodyValid;
+  };
+
+  const handleSubmit = (event: React.FormEvent<HTMLFormElement>) => {
+    event.preventDefault();
+
+    if (validateForm()) {
+      setIsAddingComment(true);
+    }
```

Comment:
The validation logic is duplicated across individual field validation functions and the `validateForm` function. Consider refactoring to use `validateForm` directly to reduce redundancy.